### ISSUE TITLE
Resource directories are not copied

### ISSUE THREAD

**Issue by detached** [spotify/docker-maven-plugin#146]
Since version 0.3.0 it is not possible to copy whole resource directories to the build directory.

This doesn't work any more:
```xml
<resource>
    <targetPath>/data</targetPath>
    <directory>${project.basedir}/data<directory>
</resource>
```

The problem was introduced with 62eaa8a8fda7200fd041387400894d908a9f975c

https://github.com/spotify/docker-maven-plugin/blob/master/src/main/java/com/spotify/docker/BuildMojo.java#L640
```java
if (copyWholeDir) {
        copiedPaths.add(separatorsToUnix(targetPath));
 } else {
        for (String included : includedFiles) {
          final Path sourcePath = Paths.get(resource.getDirectory(), included);
          final Path destPath = Paths.get(destination, targetPath, included);
          getLog().info(String.format("Copying %s -> %s", sourcePath, destPath));
          // ensure all directories exist because copy operation will fail if they don't
          Files.createDirectories(destPath.getParent());
          Files.copy(sourcePath, destPath, StandardCopyOption.REPLACE_EXISTING,
                     StandardCopyOption.COPY_ATTRIBUTES);

          copiedPaths.add(separatorsToUnix(Paths.get(targetPath, included).toString()));
     }
}
```

If `copyWholeDir` was set, the logic for copy Files will be skipped.

**Comment by detached** [comment]
As temporary workaround one can use version 0.2.13.

**Comment by Dimes** [comment]
0.2.13 doesn't have the correct behavior either. When a resource directory is copied, the resulting Dockerfile should look like this:

    ADD /local-path/ /docker-path/

Instead, version 0.2.13 adds an `ADD` statement for each file in the directory. This is a problem because each `ADD` statement creates an intermediate container, and there can only be 127 of those for any given docker image. So you're limited to adding 127 files.

**Comment by dawidmalina** [comment]
What I can say. My original PR that can be checked here https://github.com/oberthur/docker-maven-plugin/blob/master/src/main/java/com/spotify/docker/BuildMojo.java works. During PR review maintainers has decided to propose many optimizations to my code and one of this broke this functionality.

**Comment by mattnworb** [comment]
I think #157 fixes this and #146, @dawidmalina if you don't mind taking a look at that PR I'd appreciate any review.

**Comment by dawidmalina** [comment]
:+1:
